[ews-build.webkit.org] Use Twisted requests in CloseBug

JonWBedard opened the pull request:
#### 48a6464d55d55fde3d0de9cb709cd7ac671c045a
<pre>
[ews-build.webkit.org] Use Twisted requests in CloseBug
<a href="https://bugs.webkit.org/show_bug.cgi?id=251279">https://bugs.webkit.org/show_bug.cgi?id=251279</a>
rdar://104751322

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(BugzillaMixin.close_bug): Use Twisted&apos;s asynchronous requests.
(CloseBug.run): Renamed from start.
(CloseBug.start): Renamed to run.
* Tools/CISupport/ews-build/twisted_additions.py:
(TwistedAdditions.request): Automatically set Content-Type if json is specified.

Canonical link: <a href="https://commits.webkit.org/259722@main">https://commits.webkit.org/259722@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d326becab87edcc8523e9ddae21f75524c3edaf6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/6/builds/105755 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/14839 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/38619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/114967 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/175099 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/11/builds/109657 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/16248 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/85/builds/6038 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/98001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114761 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/19/builds/111507 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/16248 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/38619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/98001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/16248 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/38619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/98001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/8106 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/38619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/8593 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/85/builds/6038 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [❌ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/104552 "Failed to checkout and rebase branch from PR 9241") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/14219 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/38619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/10149 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3598 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1533,17 +1533,22 @@ def set_cq_minus_flag_on_patch(self, patch_id):
-            self._addToLog('stdio', 'Error in closing bug {}'.format(bug_id))
-            return FAILURE
-        return SUCCESS
+            yield self._addToLog('stdio', 'Error in closing bug {}'.format(bug_id))
```
Might be a good idea to also print the exception here.

aj062 reviewed (approved):
r+ assuming this is tested

JonWBedard committed:
[ews-build.webkit.org] Use Twisted requests in CloseBug
https://bugs.webkit.org/show_bug.cgi?id=251279
rdar://104751322

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(BugzillaMixin.close_bug): Use Twisted's asynchronous requests.
(CloseBug.run): Renamed from start.
(CloseBug.start): Renamed to run.
* Tools/CISupport/ews-build/twisted_additions.py:
(TwistedAdditions.request): Automatically set Content-Type if json is specified.

Canonical link: https://commits.webkit.org/259722@main
